pkg/server/handler_test.go

```diff
@@ -0,0 +1,309 @@
+package server
+
+import (
+	"testing"
+
+	"github.com/CESARBR/knot-babeltower/pkg/logging"
+	"github.com/CESARBR/knot-babeltower/pkg/network"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/controllers"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/mocks"
+)
+
+func TestMsgHandler_Start(t *testing.T) {
+	type fields struct {
+		logger          logging.Logger
+		amqp            network.AmqpReceiver
+		thingController controllers.ThingController
+	}
+	type args struct {
+		started chan bool
+		msgChan chan network.InMsg
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			"when started channel nil start should return error",
+			fields{
+				&mocks.FakeLogger{},
+				&mocks.FakeAmqpReceiver{},
+				&mocks.FakeController{},
+			},
+			args{
+				nil,
+				make(chan network.InMsg),
+			},
+			true,
+		},
+		{
+			"when msg channel nil start should return error",
+			fields{
+				&mocks.FakeLogger{},
+				&mocks.FakeAmqpReceiver{},
+				&mocks.FakeController{},
+			},
+			args{
+				make(chan bool, 1),
+				nil,
+			},
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mc := &MsgHandler{
+				logger:          tt.fields.logger,
+				amqp:            tt.fields.amqp,
+				thingController: tt.fields.thingController,
+			}
+			err := mc.start(tt.args.started, tt.args.msgChan)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("msgClientPublisher.SendRegisteredDevice() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestMsgHandler_onMsgReceived(t *testing.T) {
```

Comment:
Couldn't it be just `TestOnMsgReceived`?
There are some conventions when naming golang test functions (also applied to some variable names) as you can see [here](https://ieftimov.com/post/testing-in-go-naming-conventions/).

Reply:
Done